include/SFML/Network/Socket.hpp

```diff
@@ -31,13 +31,37 @@
 
 #include <SFML/Network/SocketHandle.hpp>
 
+#include <algorithm>
+#include <array>
+#include <cstdint>
 #include <vector>
 
 
 namespace sf
 {
 class SocketSelector;
 
+////////////////////////////////////////////////////////////
+/// \brief Robust endianness conversion
+/// \see https://mklimenko.github.io/english/2018/08/22/robust-endian-swap/
+///
+////////////////////////////////////////////////////////////
+template <typename T>
+T swapEndian(T& val, typename std::enable_if<std::is_arithmetic<T>::value, std::nullptr_t>::type = nullptr)
```

Comment:
Why a mutable l-value reference? Why not just copy it? We're only talking about types up to 8 bytes so it's not like that copy is any worse than taking a reference.